temporalcli/commands.batch.go

```diff
@@ -0,0 +1,159 @@
+		StartTime time.Time
+		CloseTime time.Time
+	}
+)
+
+func (c TemporalBatchDescribeCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resp, err := cl.WorkflowService().DescribeBatchOperation(cctx, &workflowservice.DescribeBatchOperationRequest{
+		Namespace: c.Parent.Namespace,
+		JobId:     c.JobId,
+	})
+	var notFound *serviceerror.NotFound
+	if errors.As(err, &notFound) {
+		return fmt.Errorf("could not find Batch Job '%v'", c.JobId)
+	} else if err != nil {
+		return fmt.Errorf("failed to describe batch job: %w", err)
+	}
+
+	if cctx.JSONOutput {
+		return cctx.Printer.PrintStructured(resp, printer.StructuredOptions{})
+	}
+
+	_ = cctx.Printer.PrintStructured(batchDescribe{
+		Type:           resp.OperationType.String(),
+		State:          resp.State.String(),
+		StartTime:      resp.StartTime.AsTime(),
+		CloseTime:      resp.CloseTime.AsTime(),
+		CompletedCount: fmt.Sprintf("%d/%d", resp.CompleteOperationCount, resp.TotalOperationCount),
+		FailureCount:   fmt.Sprintf("%d/%d", resp.FailureOperationCount, resp.TotalOperationCount),
+	}, printer.StructuredOptions{})
+
+	return nil
+}
+
+func (c TemporalBatchListCommand) run(cctx *CommandContext, args []string) error {
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	pageFetcher := c.pageFetcher(cctx, cl)
+	var nextPageToken []byte
+	var jobsProcessed int
+	for pageIndex := 0; ; pageIndex++ {
+		page, err := pageFetcher(nextPageToken)
+		if err != nil {
+			return fmt.Errorf("failed to list batch jobs: %w", err)
+		}
+
+		if pageIndex == 0 && len(page.GetOperationInfo()) == 0 {
+			if cctx.JSONOutput {
+				_ = cctx.Printer.PrintStructured([]any{}, printer.StructuredOptions{})
+			} else {
+				cctx.Printer.Printlnf("no Batch Jobs were found")
+			}
+			return nil
+		}
+
+		var textTable []batchTableRow
+		for _, job := range page.GetOperationInfo() {
+			if c.Limit > 0 && jobsProcessed >= c.Limit {
+				break
+			}
+			jobsProcessed++
+			// For JSON we are going to dump one line of JSON per execution
+			if cctx.JSONOutput {
+				_ = cctx.Printer.PrintStructured(job, printer.StructuredOptions{})
+			} else {
+				// For non-JSON, we are doing a table for each page
+				textTable = append(textTable, batchTableRow{
+					JobId:     job.JobId,
+					State:     job.State.String(),
+					StartTime: job.StartTime.AsTime(),
+					CloseTime: job.CloseTime.AsTime(),
+				})
+			}
+		}
+		// Print table, headers only on first table
+		if len(textTable) > 0 {
+			_ = cctx.Printer.PrintStructured(textTable, printer.StructuredOptions{
+				Table: &printer.TableOptions{NoHeader: pageIndex > 0},
+			})
+		}
+		// Stop if next page token non-existing or list reached limit
+		nextPageToken = page.GetNextPageToken()
+		if len(nextPageToken) == 0 || (c.Limit > 0 && jobsProcessed >= c.Limit) {
+			return nil
+		}
```

Comment:
There's a potential abstraction here as this code is very similar to `workflow list`. Decided against a refactor because < 3 examples is often not enough to do a good job.